integration-tests/redis-client/src/main/java/io/quarkus/redis/it/RedisWithNamedClientResource.java

```diff
@@ -18,13 +18,15 @@
 @Path("/quarkus-redis-with-named")
 @ApplicationScoped
 public class RedisWithNamedClientResource {
-    @Inject
-    @RedisClientName("named-client")
-    RedisClient redisClient;
+    private RedisClient redisClient;
+    private ReactiveRedisClient reactiveRedisClient;
 
     @Inject
-    @RedisClientName("named-client")
-    ReactiveRedisClient reactiveRedisClient;
+    public RedisWithNamedClientResource(@RedisClientName("named-client") ReactiveRedisClient reactiveRedisClient,
+            @RedisClientName("named-client") RedisClient redisClient) {
+        this.reactiveRedisClient = reactiveRedisClient;
+        this.redisClient = redisClient;
+    }
```

Comment:
I would add a specific test rather than fiddling with an existing one.

Reply:
I've changed it the way @cescoffier suggested here https://github.com/quarkusio/quarkus/pull/12902#discussion_r510853452 with constructor and field injection.

Reply:
It's really better to add an entirely separate test, testing both clients for each injection type.

Just copy/paste the existing one and change the injection.